### PR TITLE
#3814 Replacement of CacheEntryFilter by java.util.function.Predicate

### PR DESCRIPTION
**Pull request by marcelmu**
(unfortunately no JUnit tests for Job class)

### CODE REVIEW

File: modules/dcache/src/main/java/org/dcache/pool/migration/NotFilter.java [hunk 1]
```diff
@@ -1,23 +1,24 @@
...
 
+import java.util.function.Predicate;
 import org.dcache.pool.repository.CacheEntry;
 
 /**
  * Repository entry filter wrapping another filter. It accepts all
  * entries not accepted by the other filter.
  */
-public class NotFilter implements CacheEntryFilter
+public class NotFilter implements Predicate<CacheEntry>
```
Comment (kofemann) [inline modules/dcache/src/main/java/org/dcache/pool/migration/NotFilter.java]: As you use standard Predicate, then there are no reasons to keep NotFilter(which is not used anyway)

**Comment by kofemann** [comment]
ok to test

File: modules/dcache/src/main/java/org/dcache/pool/migration/NotFilter.java (hunk 1 above)
Comment (marcelmu) [reply modules/dcache/src/main/java/org/dcache/pool/migration/NotFilter.java]: I cleaned up the classes in the second commit of this pull request and removed NotStickyOwnerFilter.
I indeed overlooked NotFilter.

Comment (marcelmu) [reply modules/dcache/src/main/java/org/dcache/pool/migration/NotFilter.java]: Fixed in 1fcf3a53b955616eb7c06c1e7a5b880880dd96f8

**Comment by kofemann** [comment]
thanks!

**Pull request merged**